dll/win32/newdev/newdev.c

```diff
@@ -353,6 +355,10 @@ SearchDriverRecursive(
     BOOL retval = FALSE;
     HANDLE hFindFile = INVALID_HANDLE_VALUE;
 
+    /* Path with pattern is too long to be searched */
+    if (wcslen(Path) + 2 >= _countof(PathWithPattern))
+        return FALSE;
+
     wcscpy(DirPath, Path);
 
     if (DirPath[wcslen(DirPath) - 1] != '\\')
```

Comment:
Code change suggestion for lines 358 to 368 (range that I cannot fully select in the github interface):
```suggestion
    hr = StringCchPrintfW(DirPath, _countof(DirPath), L"%s%c",
                          Path, Path[wcslen(Path) - 1] != '\\' ? L'\\' : L'\0');
    if (!SUCCEEDED(hr))
        return FALSE;  /* Path with pattern is too long to be searched */
    hr = StringCchPrintfW(PathWithPattern, _countof(PathWithPattern), L"%s*", DirPath);
    if (!SUCCEEDED(hr))
        return FALSE;  /* Path with pattern is too long to be searched */
```

Reply:
Seems to be a lot of changes to only adding 2 characters to a known length variable.  

> less code – less bugs

I propose this change more simple:

```
WCHAR DirPath[MAX_PATH + 1];
WCHAR PathWithPattern[MAX_PATH + 2];
```
Like this, if Path < MAX_PATH, DirPath and PathWithPattern can't be overflow.